crates/ruma-common/src/identifiers/key_id.rs

```diff
@@ -33,12 +66,30 @@ impl<A: KeyAlgorithm, K: KeyName + ?Sized> KeyId<A, K> {
         Self::from_borrowed(&res).to_owned()
     }
 
-    /// Returns key algorithm of the key ID.
+    /// Returns key algorithm of the key ID - the part that comes before the colon.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use ruma_common::{KeyId, OwnedDeviceKeyId};
+    ///
+    /// let k: OwnedDeviceKeyId = KeyId::parse("ed25519:1").unwrap();
+    /// assert_eq!(k.algorithm().as_str(), "ed25519");
```

Comment:
I don't think we should convert it back to a string but check equality with the enum variant on the right side instead.

Reply:
Done in 23854e6f